src/cpp/transport/tcp/RTCPMessageManager.cpp

```diff
@@ -491,56 +511,56 @@ void RTCPMessageManager::processCheckLogicalPortsRequest(
         SerializedPayload_t payload(static_cast<uint32_t>(
             CheckLogicalPortsResponse_t::getBufferCdrSerializedSize(response)));
         response.serialize(&payload);
-        sendData(p_channel_resource, CHECK_LOGICAL_PORT_RESPONSE, transaction_id, &payload, RETCODE_OK);
+        sendData(channel, CHECK_LOGICAL_PORT_RESPONSE, transaction_id, &payload, RETCODE_OK);
     }
 }
 
 ResponseCode RTCPMessageManager::processKeepAliveRequest(
-        TCPChannelResource *p_channel_resource,
+        std::shared_ptr<TCPChannelResource>& channel,
```

Comment:
Using a reference instead of a copy can cause problems as the channel being freed while we are still using it.